### PR TITLE
Np 9334 course access checker

### PR DESCRIPTION
**Pull request by truhacevkir**

### CODE REVIEW

File: cognito-pre-token-generation-openid/src/main/java/no/unit/nva/cognito/UserSelectionUponLoginHandler.java [hunk 1]
```diff
@@ -78,41 +82,39 @@ public UserSelectionUponLoginHandler(CognitoIdentityProviderClient cognitoClient
         this.customerService = customerService;
         this.userCreator = new UserEntriesCreatorForPerson(customerService, cristinClient, identityService);
     }
-
+    
     @Override
```
Comment (truhacevkir) [inline cognito-pre-token-generation-openid/src/main/java/no/unit/nva/cognito/UserSelectionUponLoginHandler.java:86]: Har ikke rørt denne, tror det er Ketil sine endringer

File: cognito-pre-token-generation-openid/src/main/java/no/unit/nva/cognito/UserSelectionUponLoginHandler.java [hunk 2]
```diff
@@ -78,41 +82,39 @@ public UserSelectionUponLoginHandler(CognitoIdentityProviderClient cognitoClient
...
+        var request = DescribeUserPoolClientRequest.builder()
+            .userPoolId(input.getUserPoolId())
+            .clientId(input.getCallerContext().getClientId())
+            .build();
+        final var response = cognitoClient.describeUserPoolClient(request);
+        var clientName = response.userPoolClient().clientName();
+        final long end = System.currentTimeMillis();
+        logger.info("Application client name in use: {} ({} ms)", clientName, end - start);
+        
+        loginToNva(input);
```
Comment (truhacevkir) [inline cognito-pre-token-generation-openid/src/main/java/no/unit/nva/cognito/UserSelectionUponLoginHandler.java:100]: Har ikke rørt denne, tror det er Ketil sine endringer

File: fsproxy/src/main/java/no/unit/identityservice/fsproxy/FsApi.java [hunk 3]
```diff
@@ -0,0 +1,202 @@
...
+    public static final String LIMIT_IDENTIFIER = "limit";
+    public static final String LIMIT_VALUE = "0";
+    private static final Environment ENVIRONMENT = new Environment();
+    public static final URI FS_HOST = readFsHost();
+    private final String username = ENVIRONMENT.readEnv("FS_USERNAME");
+    private final String password = ENVIRONMENT.readEnv("FS_PASSWORD");
+    private final URI baseFsHostUrl;
+
+    @SuppressWarnings("PMD")
+    public FsApi(HttpClient httpClient, URI baseFsHostUrl) {
```
Comment (ketilaa) [inline fsproxy/src/main/java/no/unit/identityservice/fsproxy/FsApi.java:50]: Hvorfor har konstruktøren en parameter `httpClient` som ikke brukes?

File: fsproxy/src/main/java/no/unit/identityservice/fsproxy/FsApi.java [hunk 4]
```diff
@@ -0,0 +1,202 @@
...
+    public List<FsCourse> getCoursesToStudent(FsIdNumber fsIdNumber) throws IOException, InterruptedException {
+
+        var response = getResponse(createSearchCourseUri(fsIdNumber));
+        var responseBody = response.body();
+        var fsCoursesSearchResult = JsonUtils.dtoObjectMapper.readValue(responseBody, FsCoursesSearchResult.class);
+
+        return fsCoursesSearchResult.getItems().stream().map(c -> c.getId().getCourse()).collect(Collectors.toList());
+    }
+
+    public Boolean getFsPersonStatus(NationalIdentityNumber nin) throws IOException, InterruptedException {
```
Comment (ketilaa) [inline fsproxy/src/main/java/no/unit/identityservice/fsproxy/FsApi.java:105]: Ser du bruker klassen Boolean og ikke bare den primitive typen boolean her. Er retur av NULL et case her? Hvis ikke. bruk primitiv type boolean.

File: fsproxy/src/main/java/no/unit/identityservice/fsproxy/model/person/NationalIdentityNumber.java [hunk 5]
```diff
@@ -0,0 +1,42 @@
+package no.unit.identityservice.fsproxy.model.person;
+
+import java.util.Objects;
+import nva.commons.core.JacocoGenerated;
+
+public class NationalIdentityNumber {
+
+    public static final int FIRST_DIGIT_OF_PERSONAL_NUMBER = 6;
```
Comment (ketilaa) [inline fsproxy/src/main/java/no/unit/identityservice/fsproxy/model/person/NationalIdentityNumber.java:8]: Disse konstantene trenger vel ikke å være public? Jeg synes ikke tester trenger å vite om dem heller. Tester bør heller teste faktiske verdier.

**Review by ketilaa — CHANGES_REQUESTED** [review]

File: fsproxy/src/main/java/no/unit/identityservice/fsproxy/FsApi.java (hunk 4 above)
Comment (truhacevkir) [reply fsproxy/src/main/java/no/unit/identityservice/fsproxy/FsApi.java]: endret

File: cognito-pre-token-generation-openid/src/main/java/no/unit/nva/cognito/UserSelectionUponLoginHandler.java [hunk 6]
```diff
@@ -78,41 +82,39 @@ public UserSelectionUponLoginHandler(CognitoIdentityProviderClient cognitoClient
...
+        
+        final var start = System.currentTimeMillis();
+        var request = DescribeUserPoolClientRequest.builder()
+            .userPoolId(input.getUserPoolId())
+            .clientId(input.getCallerContext().getClientId())
+            .build();
+        final var response = cognitoClient.describeUserPoolClient(request);
+        var clientName = response.userPoolClient().clientName();
+        final long end = System.currentTimeMillis();
+        logger.info("Application client name in use: {} ({} ms)", clientName, end - start);
```
Comment (anetteOlli) [inline cognito-pre-token-generation-openid/src/main/java/no/unit/nva/cognito/UserSelectionUponLoginHandler.java:98]: Er det en grunn til at vi skal benchmarke tidsbruken? Bare lurer

File: fsproxy/src/main/java/no/unit/identityservice/fsproxy/model/person/FsStudent.java [hunk 7]
```diff
@@ -0,0 +1,2 @@
+package no.unit.identityservice.fsproxy.model.person;
```
Comment (anetteOlli) [inline fsproxy/src/main/java/no/unit/identityservice/fsproxy/model/person/FsStudent.java:1]: Denne fila kan kanskje slettes?

Comment (truhacevkir) [reply fsproxy/src/main/java/no/unit/identityservice/fsproxy/model/person/FsStudent.java:1]: Det kan den, har ikke lagt merke til at den er tom. Fjerner den.

File: cognito-pre-token-generation-openid/src/main/java/no/unit/nva/cognito/UserSelectionUponLoginHandler.java (hunk 6 above)
Comment (truhacevkir) [reply cognito-pre-token-generation-openid/src/main/java/no/unit/nva/cognito/UserSelectionUponLoginHandler.java:98]: Ikke nei, Orestis og Ketil som testet responstiden en gang.

File: fsproxy/src/main/java/no/unit/identityservice/fsproxy/model/staffperson/FsRoleToStaffPerson.java [hunk 8]
```diff
@@ -0,0 +1,45 @@
...
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import java.util.Objects;
+import nva.commons.core.JacocoGenerated;
+
+public class FsRoleToStaffPerson {
+
+    @JsonProperty("href")
+    private final String hrefToRole;
```
Comment (axthosarouris) [inline fsproxy/src/main/java/no/unit/identityservice/fsproxy/model/staffperson/FsRoleToStaffPerson.java:12]: should be a URI

File: fsproxy/src/main/java/no/unit/identityservice/fsproxy/model/staffperson/FsUriToCourseActivity.java [hunk 9]
```diff
@@ -0,0 +1,40 @@
...
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import java.util.Objects;
+import nva.commons.core.JacocoGenerated;
+
+public class FsUriToCourseActivity {
+
+    @JsonProperty("href")
+    private final String uriToCourse;
```
Comment (axthosarouris) [inline fsproxy/src/main/java/no/unit/identityservice/fsproxy/model/staffperson/FsUriToCourseActivity.java:11]: Should be a URI

File: fsproxy/src/test/java/no/unit/identityservice/fsproxy/FsApiTest.java [hunk 10]
```diff
@@ -0,0 +1,50 @@
...
+
+    @Test
+    void shouldThrowExceptionWhenPersonIsNotFoundInFs() {
+        var personNotInFs = fsMock.createResponseForPersonNotInFs();
+
+        assertThrows(UserPrincipalNotFoundException.class, () -> fsApi.getCourses(personNotInFs));
+    }
+
+    @Test
+    void shouldReturnAllCoursesToPerson() throws IOException, InterruptedException {
```
Comment (axthosarouris) [inline fsproxy/src/test/java/no/unit/identityservice/fsproxy/FsApiTest.java:40]: 4 cases here: neither student. nor teacher, only student, only teacher, both

**Review by axthosarouris — CHANGES_REQUESTED** [review]
all he

File: fsproxy/src/main/java/no/unit/identityservice/fsproxy/FsApi.java [hunk 11]
```diff
@@ -0,0 +1,189 @@
...
+
+    public static final String FODSELSDATO_IDENTIFIER = "fodselsdato0";
+    public static final String PERSONNUMMER_IDENTIFIER = "personnummer0";
+    public static final String PERSONROLLER_PATH = "personroller";
+    public static final String SEMESTER_AR_PATH = "semester.ar";
+    public static final String UNDERVISNINGSAKTIVITETER_PATH = "undervisningsaktiviteter";
+    public static final String PERSON_PATH = "personer";
+    public static final String STUDENTUNDERVISNING_PATH = "studentundervisning";
+    public static final String UNDERVISNING_SEMESTER_AR_PATH = "undervisning.semester.ar";
+    public static final String PERSON_PERSONLOPENUMMER_PATH = "person.personlopenummer";
```
[the remark this answers is not in the document]
Comment (truhacevkir) [reply fsproxy/src/main/java/no/unit/identityservice/fsproxy/FsApi.java:41]: +

File: fsproxy/src/main/java/no/unit/identityservice/fsproxy/FsApi.java [hunk 12]
```diff
@@ -0,0 +1,189 @@
...
+                   .addQueryParameter(SEMESTER_AR_PATH, year)
+                   .getUri();
+    }
+
+    private URI createSearchCourseToRoleUri(String href) {
+        return UriWrapper.fromUri(baseFsHostUrl).addChild(PERSONROLLER_PATH).addChild(href).getUri();
+    }
+
+    private URI createSearchCourseForStaffPersonUri(String href) {
+        return UriWrapper.fromUri(baseFsHostUrl).addChild(UNDERVISNINGSAKTIVITETER_PATH).addChild(href).getUri();
```
[the remark this answers is not in the document]
Comment (truhacevkir) [reply fsproxy/src/main/java/no/unit/identityservice/fsproxy/FsApi.java:181]: +

File: fsproxy/src/main/java/no/unit/identityservice/fsproxy/model/course/FsCourse.java [hunk 13]
```diff
@@ -0,0 +1,58 @@
...
+
+    @JsonProperty("emne")
+    final FsSubject subject;
+    @JsonProperty("terminnummer")
+    final int termNumber;
+    @JsonProperty("semester")
+    final FsSemester semester;
+
+    @JsonCreator
+    public FsCourse(@JsonProperty("emne") FsSubject subject, @JsonProperty("terminnummer") int termNumber,
```
[the remark this answers is not in the document]
Comment (truhacevkir) [reply fsproxy/src/main/java/no/unit/identityservice/fsproxy/model/course/FsCourse.java:20]: +

File: fsproxy/src/main/java/no/unit/identityservice/fsproxy/model/course/FsCourse.java [hunk 14]
```diff
@@ -0,0 +1,58 @@
...
+import java.util.Objects;
+import nva.commons.core.JacocoGenerated;
+
+@SuppressWarnings("PMD")
+
+public class FsCourse {
+
+    @JsonProperty("emne")
+    final FsSubject subject;
+    @JsonProperty("terminnummer")
```
[the remark this answers is not in the document]
Comment (truhacevkir) [reply fsproxy/src/main/java/no/unit/identityservice/fsproxy/model/course/FsCourse.java:14]: +

File: fsproxy/src/main/java/no/unit/identityservice/fsproxy/model/staffperson/FsRoleToStaffPerson.java (hunk 8 above)
Comment (truhacevkir) [reply fsproxy/src/main/java/no/unit/identityservice/fsproxy/model/staffperson/FsRoleToStaffPerson.java:12]: +

File: fsproxy/src/main/java/no/unit/identityservice/fsproxy/model/staffperson/FsUriToCourseActivity.java (hunk 9 above)
Comment (truhacevkir) [reply fsproxy/src/main/java/no/unit/identityservice/fsproxy/model/staffperson/FsUriToCourseActivity.java:11]: +

File: fsproxy/src/test/java/no/unit/identityservice/fsproxy/FsApiTest.java (hunk 10 above)
Comment (truhacevkir) [reply fsproxy/src/test/java/no/unit/identityservice/fsproxy/FsApiTest.java:40]: +

File: cognito-pre-token-generation-openid/src/main/java/no/unit/nva/cognito/UserSelectionUponLoginHandler.java (hunk 6 above)
Comment (truhacevkir) [reply cognito-pre-token-generation-openid/src/main/java/no/unit/nva/cognito/UserSelectionUponLoginHandler.java]: done

File: customer-commons/build.gradle [hunk 15]
```diff
@@ -15,6 +15,7 @@ dependencies {
     implementation libs.aws.sdk2.dynamodb
     implementation libs.aws.sdk2.urlconnection
 
+    testImplementation 'io.github.ganadist.sqlite4java:libsqlite4java-osx-aarch64:1.0.392'
```
Comment (axthosarouris) [inline customer-commons/build.gradle:18]: is this necessary? everything builds without it.

File: fsproxy/src/main/java/no/unit/identityservice/fsproxy/FsApi.java [hunk 16]
```diff
@@ -0,0 +1,194 @@
...
+                                 .map(attempt(this::getCourseToStaffPersonGivenUriToCourse))
+                                 .map(Try::orElseThrow)
+                                 .collect(Collectors.toList());
+
+        var coursesIfStudent = getCoursesToStudent(fsId);
+
+        var allCourses = Stream.concat(coursesIfStaff.stream(), coursesIfStudent.stream())
+                             .collect(Collectors.toList());
+
+        if (allCourses == null) {
```
Comment (axthosarouris) [inline fsproxy/src/main/java/no/unit/identityservice/fsproxy/FsApi.java:82]: allCourses == null is always false

File: fsproxy/src/main/java/no/unit/identityservice/fsproxy/FsApi.java [hunk 17]
```diff
@@ -0,0 +1,194 @@
...
+        this(HttpClient.newBuilder().build(), FS_HOST);
+    }
+
+    @JacocoGenerated
+    public List<FsCourse> getCourses(NationalIdentityNumber nin) throws IOException, InterruptedException {
+        var fsId = getFsId(nin);
+
+        var roles = getRolesToStaffPerson(fsId);
+
+        var coursesUri = roles.stream()
```
Comment (axthosarouris) [inline fsproxy/src/main/java/no/unit/identityservice/fsproxy/FsApi.java:67]: Why are you collecting to a list only to generate a stream again?

File: fsproxy/src/main/java/no/unit/identityservice/fsproxy/FsApi.java [hunk 18]
```diff
@@ -0,0 +1,194 @@
...
+        this.httpClient = httpClient;
+    }
+
+    @JacocoGenerated
+    public FsApi() {
+        this(HttpClient.newBuilder().build(), FS_HOST);
+    }
+
+    @JacocoGenerated
+    public List<FsCourse> getCourses(NationalIdentityNumber nin) throws IOException, InterruptedException {
```
Comment (axthosarouris) [inline fsproxy/src/main/java/no/unit/identityservice/fsproxy/FsApi.java:62]: Suggestion:
```
@JacocoGenerated
    public List<FsCourse> fetchCoursesForPerson(NationalIdentityNumber nin) throws IOException, InterruptedException {
        var fsId = getFsId(nin);
        var coursesIfStaff = fetchCoursesForTeachingEmployees(fsId);
        var coursesIfStudent = getCoursesToStudent(fsId);
        return Stream.concat(coursesIfStaff.stream(), coursesIfStudent.stream()).collect(Collectors.toList());
    }
    
    private List<FsCourse> fetchCoursesForTeachingEmployees(FsIdNumber fsId) throws IOException, InterruptedException {
        var roles = getRolesToStaffPerson(fsId);
        var coursesUri = fetchAllCourseUrisBasedOnRolesOfEmployee(roles);
        return fetchCourseDetails(coursesUri);
    }
    
    private List<FsCourse> fetchCourseDetails(Stream<FsUriToCourseActivity> coursesUri) {
        return coursesUri
                   .map(attempt(this::getCourseToStaffPersonGivenUriToCourse))
                   .map(Try::orElseThrow)
                   .collect(Collectors.toList());
    }
    
    private Stream<FsUriToCourseActivity> fetchAllCourseUrisBasedOnRolesOfEmployee(List<FsRoleToStaffPerson> roles) {
        return roles.stream()
                   .map(attempt(this::getCourseActivityUriToGivenRole))
                   .map(Try::orElseThrow);
    }
```

File: fsproxy/src/main/java/no/unit/identityservice/fsproxy/FsApi.java [hunk 19]
```diff
@@ -0,0 +1,194 @@
...
+        this.baseFsHostUrl = baseFsHostUrl;
+        this.httpClient = httpClient;
+    }
+
+    @JacocoGenerated
+    public FsApi() {
+        this(HttpClient.newBuilder().build(), FS_HOST);
+    }
+
+    @JacocoGenerated
```
Comment (axthosarouris) [inline fsproxy/src/main/java/no/unit/identityservice/fsproxy/FsApi.java:61]: Why JacocoGenerated?

File: fsproxy/src/main/java/no/unit/identityservice/fsproxy/model/course/FsCourse.java [hunk 20]
```diff
@@ -0,0 +1,51 @@
+package no.unit.identityservice.fsproxy.model.course;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import java.util.Objects;
+import nva.commons.core.JacocoGenerated;
+
+@SuppressWarnings("PMD")
```
Comment (axthosarouris) [inline fsproxy/src/main/java/no/unit/identityservice/fsproxy/model/course/FsCourse.java:8]: why all?

File: fsproxy/src/main/java/no/unit/identityservice/fsproxy/model/course/FsCourse.java [hunk 21]
```diff
@@ -0,0 +1,51 @@
...
+
+public class FsCourse {
+
+    @JsonProperty("emne")
+    final FsSubject subject;
```
Comment (axthosarouris) [inline fsproxy/src/main/java/no/unit/identityservice/fsproxy/model/course/FsCourse.java:13]: they should be private

File: fsproxy/src/test/java/no/unit/identityservice/fsproxy/FsMock.java [hunk 22]
```diff
@@ -0,0 +1,462 @@
...
+        }
+
+        List<FsCourseActivity> coursesToStaffPerson = uriToCourseActivities.stream()
+                                                          .map(this::createCourseActivity)
+                                                          .collect(Collectors.toList());
+        for (FsUriToCourseActivity uriToCourseActivity : uriToCourseActivities) {
+            for (FsCourseActivity toStaffPerson : coursesToStaffPerson) {
+                courseActivities.put(uriToCourseActivity, toStaffPerson);
+            }
+        }
```
Comment (axthosarouris) [inline fsproxy/src/test/java/no/unit/identityservice/fsproxy/FsMock.java:120]: refactor to method and reuse

File: fsproxy/src/test/java/no/unit/identityservice/fsproxy/FsMock.java [hunk 23]
```diff
@@ -0,0 +1,462 @@
...
```
Comment (axthosarouris) [inline fsproxy/src/test/java/no/unit/identityservice/fsproxy/FsMock.java:159]: Is this used?

File: fsproxy/src/test/java/no/unit/identityservice/fsproxy/FsMock.java [hunk 24]
```diff
@@ -0,0 +1,462 @@
...
+import no.unit.identityservice.fsproxy.model.person.FsPersonSearchResponse;
+import no.unit.identityservice.fsproxy.model.person.NationalIdentityNumber;
+import no.unit.identityservice.fsproxy.model.staffperson.FsCourseActivity;
+import no.unit.identityservice.fsproxy.model.staffperson.FsRoleToStaffPerson;
+import no.unit.identityservice.fsproxy.model.staffperson.FsRolesToPersonSearchResult;
+import no.unit.identityservice.fsproxy.model.staffperson.FsUriToCourseActivity;
+import no.unit.identityservice.fsproxy.model.staffperson.FsUriToCourseActivityContainer;
+import no.unit.nva.stubs.WiremockHttpClient;
+
+public class FsMock {
```
Comment (axthosarouris) [inline fsproxy/src/test/java/no/unit/identityservice/fsproxy/FsMock.java:41]: There is a lot of unused code in this class

**Review by axthosarouris — CHANGES_REQUESTED** [review]

File: customer-commons/build.gradle (hunk 15 above)
Comment (truhacevkir) [reply customer-commons/build.gradle]: removed

File: fsproxy/src/main/java/no/unit/identityservice/fsproxy/model/course/FsCourse.java (hunk 21 above)
Comment (truhacevkir) [reply fsproxy/src/main/java/no/unit/identityservice/fsproxy/model/course/FsCourse.java:13]: fixed

File: fsproxy/src/main/java/no/unit/identityservice/fsproxy/model/course/FsCourse.java (hunk 20 above)
Comment (truhacevkir) [reply fsproxy/src/main/java/no/unit/identityservice/fsproxy/model/course/FsCourse.java:8]: fixed

File: fsproxy/src/main/java/no/unit/identityservice/fsproxy/FsApi.java (hunk 18 above)
Comment (truhacevkir) [reply fsproxy/src/main/java/no/unit/identityservice/fsproxy/FsApi.java:62]: thank you, is changed

File: fsproxy/src/main/java/no/unit/identityservice/fsproxy/FsApi.java (hunk 16 above)
Comment (truhacevkir) [reply fsproxy/src/main/java/no/unit/identityservice/fsproxy/FsApi.java:82]: fixed

File: fsproxy/src/test/java/no/unit/identityservice/fsproxy/FsMock.java (hunk 22 above)
Comment (truhacevkir) [reply fsproxy/src/test/java/no/unit/identityservice/fsproxy/FsMock.java:120]: done

File: fsproxy/src/test/java/no/unit/identityservice/fsproxy/FsMock.java (hunk 24 above)
Comment (truhacevkir) [reply fsproxy/src/test/java/no/unit/identityservice/fsproxy/FsMock.java:41]: fixed

File: fsproxy/src/test/java/no/unit/identityservice/fsproxy/FsMock.java (hunk 23 above)
Comment (truhacevkir) [reply fsproxy/src/test/java/no/unit/identityservice/fsproxy/FsMock.java]: fixed, yes it is

File: fsproxy/src/main/java/no/unit/identityservice/fsproxy/FsApi.java (hunk 17 above)
Comment (truhacevkir) [reply fsproxy/src/main/java/no/unit/identityservice/fsproxy/FsApi.java]: +

File: fsproxy/src/test/java/no/unit/identityservice/fsproxy/FsMock.java [hunk 25]
```diff
@@ -0,0 +1,398 @@
...
+    private final Map<FsIdNumber, List<FsRoleToStaffPerson>> personRoles;
+    private final Map<NationalIdentityNumber, List<FsCourse>> coursesForStudents;
+    private final Map<FsUriToCourseActivity, FsCourseActivity> courseActivities;
+    private final Map<FsRoleToStaffPerson, FsUriToCourseActivity> coursesToRoles;
+
+    private WireMockServer server;
+    private URI fsHostUri;
+    private HttpClient httpClient;
+
+    public FsMock() {
```
Comment (anetteOlli) [inline fsproxy/src/test/java/no/unit/identityservice/fsproxy/FsMock.java:66]: This class is rather large and it does TWO things:
It generates / manages mock data AND it creates stub endpoints. You should consider splitting it into two classes.

File: fsproxy/src/test/java/no/unit/identityservice/fsproxy/FsApiTest.java [hunk 26]
```diff
@@ -0,0 +1,79 @@
...
+
+    @AfterEach
+    public void tearDown() {
+        fsMock.shutDown();
+    }
+
+    @BeforeEach
+    void init() {
+        fsMock = new FsMock();
+        fsMock.initialize();
```
Comment (anetteOlli) [inline fsproxy/src/test/java/no/unit/identityservice/fsproxy/FsApiTest.java:28]: The initialize() method should be called inside the constructor of FsMock().

File: fsproxy/src/main/java/no/unit/identityservice/fsproxy/FsApi.java [hunk 27]
```diff
@@ -0,0 +1,189 @@
...
+import no.unit.identityservice.fsproxy.model.staffperson.FsCourseActivity;
+import no.unit.identityservice.fsproxy.model.staffperson.FsRoleToStaffPerson;
+import no.unit.identityservice.fsproxy.model.staffperson.FsRolesToPersonSearchResult;
+import no.unit.identityservice.fsproxy.model.staffperson.FsUriToCourseActivity;
+import no.unit.identityservice.fsproxy.model.staffperson.FsUriToCourseActivityContainer;
+import nva.commons.core.Environment;
+import nva.commons.core.JacocoGenerated;
+import nva.commons.core.attempt.Try;
+import nva.commons.core.paths.UriWrapper;
+
```
Comment (anetteOlli) [inline fsproxy/src/main/java/no/unit/identityservice/fsproxy/FsApi.java:30]: We don't need any log messages at all?

File: fsproxy/src/main/java/no/unit/identityservice/fsproxy/FsApi.java [hunk 28]
```diff
@@ -0,0 +1,189 @@
...
+    public static final String DB_IDENTIFIER = "dbId";
+    public static final String LIMIT_IDENTIFIER = "limit";
+    public static final String LIMIT_VALUE = "0";
+    private static final Environment ENVIRONMENT = new Environment();
+    public static final URI FS_HOST = readFsHost();
+    private final String username = ENVIRONMENT.readEnv("FS_USERNAME");
+    private final String password = ENVIRONMENT.readEnv("FS_PASSWORD");
+    private final URI baseFsHostUrl;
+    private final HttpClient httpClient;
+
```
Comment (anetteOlli) [inline fsproxy/src/main/java/no/unit/identityservice/fsproxy/FsApi.java:50]: No custom log messages required?
I am just asking. If the exceptions messages are good then we don't need more spam in the logs.

File: fsproxy/src/main/java/no/unit/identityservice/fsproxy/FsApi.java (hunk 27 above)
Comment (truhacevkir) [reply fsproxy/src/main/java/no/unit/identityservice/fsproxy/FsApi.java:30]: har lagt inn en logger

File: fsproxy/src/main/java/no/unit/identityservice/fsproxy/FsApi.java (hunk 28 above)
Comment (truhacevkir) [reply fsproxy/src/main/java/no/unit/identityservice/fsproxy/FsApi.java:50]: har lagt inn en logger

File: fsproxy/src/test/java/no/unit/identityservice/fsproxy/FsApiTest.java (hunk 26 above)
Comment (truhacevkir) [reply fsproxy/src/test/java/no/unit/identityservice/fsproxy/FsApiTest.java:28]: fixed

**Review by anetteOlli — APPROVED** [review]

**Review by axthosarouris — APPROVED** [review]

**Pull request merged**